src/components/Tasks/TaskList.tsx

```diff
@@ -9,31 +9,61 @@ import * as EteSync from 'etesync';
 
 import { List } from '../../widgets/List';
 
-import { TaskType, PimType } from '../../pim-types';
+import { TaskType, PimType, TaskTags } from '../../pim-types';
 import FormControlLabel from '@material-ui/core/FormControlLabel';
 import Checkbox from '@material-ui/core/Checkbox';
 import Divider from '@material-ui/core/Divider';
+import Grid from '@material-ui/core/Grid';
+import { Theme, withTheme } from '@material-ui/core/styles';
+
+import { useSelector } from 'react-redux';
 
 import TaskListItem from './TaskListItem';
 import QuickAdd from './QuickAdd';
+import Sidebar from './Sidebar';
+
+import { StoreState } from '../../store';
 
 const sortSelector = createSelector(
   (entries: TaskType[]) => entries,
   (entries) => entries.sort((a, b) => a.title.localeCompare(b.title))
 );
 
+const tagFilters = Object.assign(
+  {},
+  ...TaskTags.map((tag) => ({
+    [tag]: (x: TaskType) => x.tags.includes(tag),
+  }))
+);
+
+const filters = {
+  all: () => true,
+  ...tagFilters,
+};
+
 interface PropsType {
   entries: TaskType[];
   collections: EteSync.CollectionInfo[];
   onItemClick: (entry: TaskType) => void;
   onItemSave: (item: PimType, journalUid: string, originalItem?: PimType) => Promise<void>;
+  theme: Theme;
 }
 
-export default React.memo(function TaskList(props: PropsType) {
+export default React.memo(withTheme(function TaskList(props: PropsType) {
   const [showCompleted, setShowCompleted] = React.useState(false);
-  const entries = props.entries.filter((x) => showCompleted || !x.finished);
+  const settings = useSelector((state: StoreState) => state.settings.taskSettings);
+  const { filterBy } = settings;
+
+  const entries = props.entries.filter((x) => showCompleted || !x.finished).filter(filters[filterBy]);
   const sortedEntries = sortSelector(entries);
 
+  // counts tags and creates an object with shape { tag: amount }
+  // TODO: memoize
+  const tags = TaskTags.reduce((obj, tag) => ({ ...obj, [tag]: 0 }), {});
```

Comment:
This is quite hacky and very slow because it creates a lot of objects. Maybe use the javascript Map?
```
const tags = new Map<string, number>();
TaskTags.forEach((tag) => tags.set(tag, tags.get(tag, 0) + 1));
```
Then you can just use this map object below using `tags.get` instead of the `hasOwnProperty` hack.